src/registry/parseRegistry.spec.ts

```diff
@@ -46,4 +46,124 @@ describe('parseRegistry', () => {
 	});
+	it('Should correctly update the registry with an injectedRegsitry without specName', () => {
+		const assetsInfo = {};
+		const foreignAssetsInfo = {};
+		const opts = {
+			injectedRegistry: {
+				westend: {
+					'0': {
+						tokens: ['WND', 'WND2'],
+						assetsInfo,
+						foreignAssetsInfo,
+						poolPairsInfo: {},
+					},
+				},
+			},
+		};
+		const registry = parseRegistry(reg as ChainInfoRegistry, opts);
+
+		expect(registry.westend['0']).toStrictEqual({
+			tokens: ['WND', 'WND2'],
+			assetsInfo: {},
+			foreignAssetsInfo: {},
+			specName: 'westend',
+			poolPairsInfo: {},
+		});
+	});
+	it('Should correctly update the registry with an injectedRegsitry without tokens', () => {
+		const assetsInfo = {};
+		const foreignAssetsInfo = {};
+		const opts = {
+			injectedRegistry: {
+				westend: {
+					'0': {
+						assetsInfo,
+						foreignAssetsInfo,
+						poolPairsInfo: {},
+						specName: 'totoro',
+					},
+				},
+			},
+		};
+		const registry = parseRegistry(reg as ChainInfoRegistry, opts);
+
+		expect(registry.westend['0']).toStrictEqual({
+			tokens: ['WND', 'WND2'],
+			assetsInfo: {},
+			foreignAssetsInfo: {},
+			specName: 'totoro',
+			poolPairsInfo: {},
+		});
+	});
+	it('Should correctly error when a previously missing injectedRegsitry is added without a token', () => {
+		const expectedErrorMessage = `Must define the tokens property`;
+
+		const assetsInfo = {};
+		const foreignAssetsInfo = {};
+		const opts = {
+			injectedRegistry: {
+				westend: {
+					'3000': {
+						assetsInfo,
+						foreignAssetsInfo,
+						poolPairsInfo: {},
+						specName: 'totoro',
+					},
+				},
+			},
+		};
+
+		const err = () => parseRegistry(reg as ChainInfoRegistry, opts);
+		expect(err).toThrow(expectedErrorMessage);
+	});
+	it("Should correctly override the registry's token entry with an injectedRegsitry", () => {
+		const assetsInfo = {};
+		const foreignAssetsInfo = {};
+		const opts = {
+			injectedRegistry: {
+				westend: {
+					'0': {
+						tokens: ['WOP'],
+						assetsInfo,
+						foreignAssetsInfo,
+						poolPairsInfo: {},
+					},
+				},
+			},
+		};
+		const registry = parseRegistry(reg as ChainInfoRegistry, opts);
+
+		expect(registry.westend['0']).toStrictEqual({
+			tokens: ['WOP'],
+			assetsInfo: {},
+			foreignAssetsInfo: {},
+			specName: 'totoro',
+			poolPairsInfo: {},
+		});
+	});
+	it('Should correctly add a previously missing injectedRegsitry without assetsInfo', () => {
+		const foreignAssetsInfo = {};
+		const opts = {
+			injectedRegistry: {
+				rococo: {
+					'2000': {
+						foreignAssetsInfo,
+						tokens: ['TST'],
+						poolPairsInfo: {},
+						specName: 'testy',
+					},
+				},
+			},
+		};
+		const registry = parseRegistry(reg as ChainInfoRegistry, opts);
+
+		expect(registry.rococo['2000']).toStrictEqual({
+			tokens: ['TST'],
+			assetsInfo: {},
+			foreignAssetsInfo: {},
+			specName: 'testy',
+			poolPairsInfo: {},
+		});
+	});
```

Comment:
What if the current chain has 50 entries for `foreignAssetsInfo`, and we just want to add one more?

Reply:
Is the current case we would have to copy all 50 entries in memory to another object, then add our new asset then override the registries version?